### DISCUSSION TITLE
How to describe the table schema of a multi-page spreadsheet?

### DISCUSSION THREAD

**Discussion by vitorbaptista** [frictionlessdata/frictionlessdata.io#655] [Q&A] [answered]
What's the expected behaviour of a validator (say goodtables) when parsing a datapackage like:

```javascript
{
    "name": "my-data",
    "title": "My data",
    "resources": [
        {
            "name": "data",
            "path": "data.xls",  // Excel file with multiple pages
            "schema": { /* Table schema */ }
        }
    ]
}
```

I imagine the tool will validate the first page. Is there a way to define which page to validate?

The specific use case I'm thinking is on defining the table schema for goodtables.

(cc @pwalsh)

**Comment by pwalsh** [comment]
While we support excel in the underlying libraries (tabulator), and therefore also in goodtables, excel is not valid for a *Tabular Data Package*, and so I would not expect this DP to be parsed as a TDP as a given.

However, it might due to implementation. In that case, I am not sure. while gt has the ability to select a sheet, TDP has no such declaration, and one might expect it to select the first sheet.

**Comment by rufuspollock** [comment]
@vitorbaptista first as @pwalsh says that is not a valid tabular data package.

However, this is a natural case and we've already been handling this in DataHub, see this post: http://datahub.io/blog/excel-files-on-the-datahub-automated-previews-and-data-extraction

Our approach is to allow schema on Excel resources but have an extra sheet property. @anuveyatsu can probably comment in more detail.

**Comment by vitorbaptista** [comment]
Is there a place for these "extensions"? Although I understand TDP only supports CSV, spreadsheet formats are a very common use-case.

My vision for goodtables is that the TDP and table schema are the way most people will configure the validations, and the missing piece is defining the sheet in a spreadsheet. We can go ahead and add a `sheet` attribute, as @rufuspollock has done on datahub. Do you guys think it make sense to have these patterns somewhere in the specs, some kind of "example extensions" or "usage patterns"?

**Comment by Stephen-Gates** [comment]
https://frictionlessdata.io/specs/patterns/ ?

**Comment by roll** [comment, marked as answer]
It's possible with Frictionless Framework and I'm going to work on this as a pattern ([Table Dialect Spec](https://github.com/frictionlessdata/specs/issues/697)) for the specs:

```
{
    "name": "my-data",
    "title": "My data",
    "resources": [
        {
            "name": "data",
            "path": "data.xls",  // Excel file with multiple pages
            "schema": { /* Table schema */ }
            "dialect": {
               "sheet": "Sheet1"
            }
        },
        {
            "name": "data",
            "path": "data.xls",  // Excel file with multiple pages
            "schema": { /* Table schema */ }
            "dialect": {
               "sheet": "Sheet2"
            }
        }
    ]
}
```

The same approach works for SQL databases and other multitable formats.
